### PR TITLE
set enabled for codebuild module

### PR DESCRIPTION
**Pull request by mihaiplesa**
Resolves https://github.com/cloudposse/terraform-aws-ecs-codepipeline/issues/101

### CODE REVIEW

**Comment by mihaiplesa** [comment]
@joe-niland can we try tests again?

**Comment by mihaiplesa** [comment]
Ok, should work now @joe-niland.

**Comment by joe-niland** [comment]
@mihaiplesa let's check that, but I think the test is going to keep failing due to duplicate bucket name. I'm looking at fixing the test now.

**Comment by mihaiplesa** [comment]
I've tested with https://github.com/cloudposse/terraform-aws-ecs-web-app/compare/master...mihaiplesa:terraform-aws-ecs-web-app:master and setting `codebuild_enabled` to `false`.

One last possible change would be to replace `0` with `*`.

File: main.tf
Comment (joe-niland) [inline main.tf]:
```suggestion
  bucket = join("", aws_s3_bucket.default.*.id)
```

File: main.tf
Comment (joe-niland) [inline main.tf:22]:
```suggestion
  bucket = join("", aws_s3_bucket.default.*.id)
```

**Comment by mihaiplesa** [comment]
Ugh, for using `aws_s3_bucket_acl` we need AWS provider v4 and this module has `>= 2.0` so we'd break compatibility if we merge this. Similar in https://github.com/cloudposse/terraform-aws-codebuild.

Can we just merge if the only broken tests are because of the deprecation warning?

**Comment by joe-niland** [comment]
@mihaiplesa yes, we can leave the deprecation warning in for this one. If you want to revert that, let's try again.

**Comment by mihaiplesa** [comment]
@joe-niland done.

**Comment by joe-niland** [comment]
@mihaiplesa all your issues are OK now, but the tests are still failing due to a bucket name collision. I'm fixing that in this [PR](https://github.com/cloudposse/terraform-aws-ecs-codepipeline/pull/102) but needs an approval before merging.

**Comment by joe-niland** [comment]
@mihaiplesa can you please merge master and then we'll try it again?

**Comment by mihaiplesa** [comment]
@joe-niland done.

**Review by joe-niland — APPROVED** [review]

**Comment by joe-niland** [comment]
Thanks @mihaiplesa we got there in the end!

**Comment by mihaiplesa** [comment]
Thank you as well. This is where it will actually be used https://github.com/cloudposse/terraform-aws-ecs-web-app/pull/210

**Comment by nitrocode** [comment]
@joe-niland This change seems like a no-op.

For future ref: The change here passed in `enabled = module.this.enabled` but this is already passed in via the context.

https://github.com/mihaiplesa/terraform-aws-ecs-codepipeline/blob/f2b43ffedc30a7f4f7f57b59f9020b79fb3f8a83/main.tf#L239

@mihaiplesa please use the PR template.

If the above functionality does not have the intended effect without applying this change, then something else is very wrong...

**Comment by joe-niland** [comment]
@nitrocode thanks, I can see that now. 

@mihaiplesa I should have asked you which codebuild resources were created when `enabled` == false. Could you let us know if you still see the original issue.

**Comment by mihaiplesa** [comment]
Hmm I can't reproduce now which seems to be reassuring, it's been a while since this issue has been on my mind so I might be mixing things.

Also noticed now that `module.this.enabled` if only passed where there is a count or a more complex conditional.

Do we want to revert this PR?

**Pull request merged**
